### ISSUE TITLE
Images are not included

### ISSUE THREAD

**Issue by ro0sterjam** [danieldeng2/perlego-downloader#7]
Hey just wanted to say this is awesome and it works great, _for the most part_. Unfortunately all the photos have disappeared when trying to save... are there any workarounds?

**Comment by ro0sterjam** [comment]
Realized what the issue was. The timeout before printing did not give the page enough time to download all the images (currently it is set to `#pages * 100` ms). I changed this to a static 30s for my purposes to ensure all the images were downloaded before printing.

**Comment by Big592** [comment]
Sorry I'm a bit of a noob but how do you change it to a static 30s? and also where is #pages * 100 ms) because I can't seem to find that.

**Comment by Big592** [comment]
> Sorry I'm a bit of a noob but how do you change it to a static 30s? and also where is #pages * 100 ms) because I can't seem to find that.



> Realized what the issue was. The timeout before printing did not give the page enough time to download all the images (currently it is set to `#pages * 100` ms). I changed this to a static 30s for my purposes to ensure all the images were downloaded before printing.

Sorry I'm a bit of a noob but how do you change it to a static 30s? and also where is #pages * 100 ms) because I can't seem to find that.

**Comment by danieldeng2** [comment]
> Realized what the issue was. The timeout before printing did not give the page enough time to download all the images (currently it is set to `#pages * 100` ms). I changed this to a static 30s for my purposes to ensure all the images were downloaded before printing.

Thanks for testing, a few people also reported this as well but I couldn't find out why due to not being able to access Perlego. Will fix it in future versions.

**Comment by Big592** [comment]
> > Realized what the issue was. The timeout before printing did not give the page enough time to download all the images (currently it is set to `#pages * 100` ms). I changed this to a static 30s for my purposes to ensure all the images were downloaded before printing.
> 
> Thanks for testing, a few people also reported this as well but I couldn't find out why due to not being able to access Perlego. Will fix it in future versions.

Have you not got an account? I can give you one if you like

**Comment by Big592** [comment]
> > Realized what the issue was. The timeout before printing did not give the page enough time to download all the images (currently it is set to `#pages * 100` ms). I changed this to a static 30s for my purposes to ensure all the images were downloaded before printing.
> 
> Thanks for testing, a few people also reported this as well but I couldn't find out why due to not being able to access Perlego. Will fix it in future versions.
> 
> Edit: Fixed, see readme on how to use

Hi, I tried it now but it still ain't working. Any tips?

**Comment by ro0sterjam** [comment]
So I didn't actually didn't feel so comfortable with a constant/preset wait time so I ended up just removing the print command and manually printing it myself after I was confident the page was fully loaded.

But then doing that for a 800 page pdf in batches of 100 was just way too tedious so I ended up having the code open a new tab and setting the html for that new tab every 50 pages, so by the end of the function I had like 16 tabs open which all got an opportunity to fully load. Then it was trivial to just print each tab separately.

The only issue is that chrome doesn't allow you to open tabs in the background so I had to manually refocus the original tab every time a new tab opened...

Anyway, I don't suggest you do exactly as I did, since it doesn't work as smoothly as your existing code does. But I just wanted to share an approach that worked well for me.
